backend/bin/e2e-seed-new.js

```diff
@@ -0,0 +1,123 @@
+      if (err.message.includes('already exists')) {
+        console.log(`${StreamName} already exists`);
+        return;
+      }
+      throw new Error(`Could not create stream: ${err.message}`);
+    }
+  );
+};
+
+const createArchiveBucket = () => {
+  console.log('Creating archive S3 bucket for development');
+
+  const s3 = new AWS.S3({
+    endpoint: process.env.S3_ENDPOINT,
+    region: 'ap-southeast-2',
+    accessKeyId: 'FAKE',
+    secretAccessKey: 'ALSO FAKE',
+  });
+
+  return s3.createBucket({ Bucket: process.env.ARCHIVE_BUCKET }).promise().then(
+    () => console.log(`${process.env.ARCHIVE_BUCKET} created`),
+    err => { throw new Error(`Could not create bucket: ${err.message}`); }
+  );
+};
+
+const createStreamAndBucket = () => {
+  if (process.env.NODE_ENV === 'production') {
+    console.log('Not creating kinesis stream or S3 bucket. These should already be there in production.');
+    return undefined;
+  }
+  return Promise.all([createStream(), createArchiveBucket()]);
+};
+
+const fakeRes = () => {
+  const res = {
+    lastStatus: null,
+    lastJson: null,
+  };
+  res.status = status => { res.lastStatus = status; return res; };
+  res.sendStatus = status => { res.lastStatus = status; return res; };
+  res.json = json => { res.lastJson = json; return res; };
+  return res;
+};
+
+const createDefaultBranch = () => {
+  console.log('Creating default branch.');
+  const req = { body: { id: branchId, name: 'A branch' } };
+  const res = fakeRes();
+  return branchesController.createBranch(req, res)
+    .then(() => {
+      if (res.lastStatus === 200) {
+        console.log('Default branch created');
+        return Promise.resolve();
+      }
+      console.log('Failed to create default branch');
+      return Promise.reject({ status: res.lastStatus, json: res.lastJson });
+    });
+};
+
+const createDefaultAdmin = () => {
+  console.log('Creating default admin');
+  const req = { params: {}, body: { email: 'admin@rr.com', name: 'Default Super Admin', password: 'password4321' } };
+  const res = fakeRes();
+  return adminController.createAdmin(adminType.super)(req, res)
+    .then(() => {
+      if (res.lastStatus === 200) {
+        console.log(`Default admin created - username: ${req.body.email}, password: ${req.body.password}`);
+        return Promise.resolve();
+      }
+      console.log('Failed to create default admin');
+      return Promise.reject({ status: res.lastStatus, json: res.lastJson });
+    });
+};
+
+const wait = () => {
+  console.log('Waiting for a few seconds for things to be ready...');
+  return new Promise(resolve => setTimeout(resolve, 5000));
+};
+
+console.log('Running seed script');
+Promise.resolve()
+  .then(createStreamAndBucket)
+  .then(wait)
+  .then(createDefaultBranch)
+  .then(createDefaultAdmin)
+  .then(() => console.log('Seeded successfully.'))
+  .catch(error => {
+    console.error('Seed failed:', error);
+    process.exit(1);
+  });
```

Comment:
The diff between this file and `seed.js` is tiny, so we've got a lot of copy-paste here. I think that's fine for a first cut while we're still working all this stuff out, but at some point it might be good for the two files to either: 1) converge and be parameterised somehow, or 2) have common stuff pulled out into a seed-utils.js or something. I wouldn't do it yet but just something to think about.

Also I think we can go ahead and just call this `e2e-seed.js`, rather than `-new`. The old file contents are obsolete anyway, so no need to keep them around!

Reply:
In short: Yes to all your points. We should really take this out of the merge request.

e2e-see-new.js is a pure copy and paste as it is very early stages of work in progress. We are using the seed as a reference and adopt from there. Refactoring needed for sure once we can add the e2e dependencies.